WordPress/Classes/Utility/Media/MediaThumbnailDownloader.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+/// Utility class for downloading a remote thumbnail for Media items.
+///
+class MediaThumbnailDownloader {
+
+    /// The managed object context the utility should operate within and call closures via `performBlock`.
+    ///
+    let managedObjectContext: NSManagedObjectContext
+
+    init(managedObjectContext: NSManagedObjectContext) {
+        self.managedObjectContext = managedObjectContext
+    }
+
+    /// Download a thumbnail image for a Media item, if available.
+    ///
+    /// - Parameters:
+    ///   - media: The Media object.
+    ///   - preferredSize: The preferred size of the image, in points, to configure remote URLs for.
+    ///   - onCompletion: Completes if everything was successful, but nil if no image is available.
+    ///   - onError: An error was encountered either from the server or locally, depending on the Media object or blog.
+    ///
+    /// - Note: based on previous implementation in MediaService.m.
+    ///
+    func downloadThumbnail(forMedia media: Media,
+                           preferredSize: CGSize,
+                           onCompletion: @escaping (UIImage?) -> Void,
+                           onError: @escaping (Error) -> Void) {
+
+        managedObjectContext.perform {
+            var remoteURL: URL?
+            // Check if the Media item is a video or image.
+            if media.mediaType == .video {
+                // If a video, ensure there is a remoteThumbnailURL
+                guard let remoteThumbnailURL = media.remoteThumbnailURL else {
+                    // No video thumbnail available.
```

Comment:
You can get thumbnail from video files on remote servers. I recently added this for self-hosted sites on the old MediaService. Maybe this can be added on a follow up PR.